acvm-repo/acvm/tests/solver.rs

```diff
@@ -910,4 +1325,80 @@ proptest! {
         let (lhs, rhs) = prop_assert_zero_l(and_op, zero, x);
         prop_assert_eq!(lhs, rhs);
     }
+
+    #[test]
+    fn poseidon2_permutation_matches_external_impl(inputs in proptest::collection::vec(field_element(), 4)) {
+        let (result, expected_result) = run_both_poseidon2_permutations(inputs);
+        prop_assert_eq!(result, expected_result)
+    }
+
+    #[test]
+    fn sha256_injective(inputs_distinct_inputs in any_distinct_inputs(None, 0, 32)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        let (result, message) = prop_assert_injective(inputs, distinct_inputs, 32, sha256_op);
+        prop_assert!(result, "{}", message);
+    }
+
+    #[test]
+    fn sha256_compression_injective(inputs_distinct_inputs in any_distinct_inputs(None, 24, 24)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        if inputs.len() == 24 && distinct_inputs.len() == 24 {
+            let (result, message) = prop_assert_injective(inputs, distinct_inputs, 8, sha256_compression_op);
+            prop_assert!(result, "{}", message);
+        }
+    }
+
+    #[test]
+    fn blake2s_injective(inputs_distinct_inputs in any_distinct_inputs(None, 0, 32)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        let (result, message) = prop_assert_injective(inputs, distinct_inputs, 32, blake2s_op);
+        prop_assert!(result, "{}", message);
+    }
+
+    #[test]
+    fn blake3_injective(inputs_distinct_inputs in any_distinct_inputs(None, 0, 32)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        let (result, message) = prop_assert_injective(inputs, distinct_inputs, 32, blake3_op);
+        prop_assert!(result, "{}", message);
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5690): expected to be injective
+    #[test]
+    #[should_panic(expected = "not injective")]
+    fn keccak256_injective(inputs_distinct_inputs in any_distinct_inputs(Some(8), 0, 32)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        let (result, message) = prop_assert_injective(inputs, distinct_inputs, 32, keccak256_op);
+        prop_assert!(result, "{}", message);
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5689): doesn't fail with an error, returns constant output
```

Comment:
Once we fix the inputs for keccak256 we should see whether this issue still exists

Reply:
It still does: I fixed the input sizes with `max_input_bits`

Reply:
Did we verify that the bytes passed to the keccak library method are in fact different in this failing case? I still see we use `FieldElement::max_num_bits()` in `constant_or_witness_to_function_inputs`?

Reply:
Ok, I updated that other case and it's now passing 🙏